build.rs

```diff
@@ -0,0 +1,43 @@
+use std::fs::File;
+///
+/// Build Script
+/// This is run as a pre-build step -- before the rust backend is compiled.
+///
+use std::{io::Write, process::Command};
+
+/*
+ * Note 1: this file was written for *nix systems -- it likely won't
+ * work on windows!
+ */
+
+#[allow(dead_code)]
+fn shell(command: &str) {
+    // println!("build.rs => {}", command);
+
+    let output = Command::new("sh")
+        .arg("-c")
+        .arg(command)
+        .output()
+        .expect(format!("Failed to run {cmd}", cmd = command).as_str());
+
+    // println!("build.rs => {:?}", output.stdout);
+    let mut file = File::create("build.log").expect("Couldn't create file...");
+    file.write(b"build log\n\n\n\nSTDOUT:\n")
+        .expect("Couldn't write to build log");
+    file.write_all(&output.stdout)
+        .expect("Couldn't write to build log");
+    file.write(b"\n\n\n\nSTDERR:\n")
+        .expect("Couldn't write to build log");
+    file.write_all(&output.stderr)
+        .expect("Couldn't write to build log");
+}
+
+fn main() {
+    // Only install frontend dependencies when building release
+    // #[cfg(not(debug_assertions))]
+    shell("cd frontend && npm install --frozen-lockfile");
```

Comment:
I think it is better to place this logic into `build.rs` inside `frontend` directory. On the top level build.rs should only copy the results of `npm build` into the `wwwroot` directory of `backend`

Reply:
Resolved? I'm not sure what exactly `copy ... into the wwwroot directory of backend` means

Reply:
Axum can be used as a hosting for forntend files. Common name of the directory, that contains all static files (including frontend files) is `wwwroot`. So I meant, that `build.rs` can copy the build result of frontend (page, js, css files) into the folder for all static files inside the build result of backend